tests/integration/targets/ec2_ami/tasks/main.yml

```diff
@@ -628,6 +628,50 @@
           - not result.changed
           - not result.failed
 
+    # ============================================================
+
+    - name: create an image from the snapshot with boot_mode and tpm_support
+      ec2_ami:
+        name: '{{ ec2_ami_name }}_ami-boot-tpm'
+        description: '{{ ec2_ami_description }}'
+        state: present
+        boot_mode: uefi
+        tpm_support: v2.0
+        launch_permissions:
+          user_ids: []
+        tags:
+          Name: '{{ ec2_ami_name }}_ami-boot-tpm'
+        root_device_name: '{{ ec2_ami_root_disk }}'
+        device_mapping:
+          - device_name: '{{ ec2_ami_root_disk }}'
+            volume_type: gp2
+            size: 8
+            delete_on_termination: true
+            snapshot_id: '{{ setup_snapshot.snapshot_id }}'
+      register: result
+      ignore_errors: true
+      vars:
+        ansible_python_interpreter: "{{ botocore_virtualenv_interpreter }}"
+
+    - name: set image id fact for deletion later
+      set_fact:
+        ec2_ami_image_id_boot_tpm: "{{ result.image_id }}"
+        ec2_ami_snapshot_boot_tpm: "{{ result.block_device_mapping[ec2_ami_root_disk].snapshot_id }}"
+
+    - name: gather facts about the image created
+      ec2_ami_info:
+        image_ids: '{{ ec2_ami_image_id_boot_tpm }}'
+      register: ami_facts_result_boot_tpm
+      ignore_errors: true
+
+    - name: assert that new ami has been created with desired options
+      assert:
+        that:
+          - "result.changed"
+          - "result.image_id.startswith('ami-')"
+          - ami_facts_result_boot_tpm.images[0].image_id | length != 0
+          - ami_facts_result_boot_tpm.images[0].boot_mode == 'uefi'
+          - ami_facts_result_boot_tpm.images[0].tpm_support == 'v2.0'
```

Comment:
Can you also test `uefi_data`?

Reply:
trying to figure out what a valid `uefi_data_file` would look like, currently boto is throwing error on below task
```
    - name: AMI Creation with boot_mode and tpm_support
      amazon.aws.ec2_ami:
        name: newtest-uefi-file-data
        state: present
        architecture: x86_64
        virtualization_type: hvm
        root_device_name: /dev/sda1
        device_mapping:
          - device_name: /dev/sda1
            snapshot_id: snap-0f00cba7123456
        wait: yes
        region: us-east-2
        boot_mode: uefi
        uefi_data: "{{ lookup('file', 'uefi_data_file') | b64encode }}"
        tpm_support: v2.0
        tags:
          name: test-ami
```

```
botocore.exceptions.ClientError: An error occurred (InvalidParameterValue)
when calling the RegisterImage operation: Invalid uefiData.
```

Reply:
I would suggest to use `get_instance_uefi_data()` https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2.html#EC2.Client.get_instance_uefi_data to get a working uefidata file and you can send it back later in the tests.

Reply:
tried that but turns out it throws `unauthorized` error for the operator `GetInstanceUefiData`.
Would need to open a PR to get the permissions added to terminator repository.

Reply:
@goneri opened PR to add required permission.
https://github.com/mattclay/aws-terminator/pull/229

Task to get uefi data in integration tests: https://github.com/ansible-collections/amazon.aws/pull/1037/files#diff-ceba09bf3546fc4c878b0ae63333fa702ef39a1c13fb206f78333031fd67dc07R77-R86

Reply:
the permission issue has been resolved, currently trying to solve error
```
An error occurred (InvalidInstanceID.NotFound) when calling the GetInstanceUefiData operation: 
Unknown virtIds for owner 'xxxxxxx': [ i-xxxxxxxxxx ]"
```

Reply:
@goneri, it appears that there are currently no provided `AMIs` with `uefi boot mode` support for `linux`, also none of our currently [supported instance types](https://github.com/mattclay/aws-terminator/blob/master/aws/policy/compute.yaml#L24-L29) for integration tests support `uefi boot mode`.
I can test the code for `UEFI boot mode` and `UEFI data` manually but not sure if it's possible to add to `integration tests`.

Relevant links:
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/launch-instance-with-uefi-sb.html
https://docs.aws.amazon.com/AWSEC2/latest/WindowsGuide/enable-nitrotpm-prerequisites.html
https://aws.amazon.com/blogs/aws/amazon-ec2-now-supports-nitrotpm-and-uefi-secure-boot/

Reply:
This will be a good candidate for an unit-test.

Reply:
Added a unit test, review would be helpful.
https://github.com/ansible-collections/amazon.aws/pull/1037/files#diff-e73e5db91a3de8db23f7c80c49a50b547462a582d68095d4580d2170c90b30f8